test/uk/gov/hmrc/ddcopsusermanagmentfrontendpocscala/UserconnectorSpec.scala

```diff
@@ -14,66 +14,100 @@
  * limitations under the License.
  */
 
-import uk.gov.hmrc.http._
+package uk.gov.hmrc.ddcopsusermanagmentfrontendpocscala
+
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 import com.github.tomakehurst.wiremock.client.WireMock._
-import uk.gov.hmrc.http.test.{HttpClientSupport, WireMockSupport}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
 import play.api.Configuration
 import play.api.libs.functional.syntax._
 import play.api.libs.json._
-import org.scalatest.matchers.should.Matchers
-import org.scalatest.wordspec.AnyWordSpec
-import org.scalatest.concurrent.ScalaFutures
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.Future
-
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http._
+import uk.gov.hmrc.http.test.{HttpClientSupport, WireMockSupport}
 
+import scala.concurrent.{ExecutionContext, Future}
 
 case class User(title: String, firstName: String)
 
-class UserConnector(httpClient: HttpClient, configuration: Configuration) {
-    val baseUrl: String = configuration.get[String]("connector.url") //.getOrElse(throw new IllegalArgumentException("missing connector.url configuration"))
+class UserConnector(httpClient: HttpClient, configuration: Configuration)(implicit
+  executionContext: ExecutionContext
```

Comment:
for the purposes of the test we had the implicit execution context being imported in at the top of the file, but I've moved it so it's an implicit class parameter, and the import now happens inside the test suite. The implicit header carrier I've moved to the test suite too and made it get passed in as an implicit method parameter of `search()`.